### ISSUE TITLE
fortios_ips_sensor: plan / apply will always detect a change when there is none to be made

### ISSUE THREAD

**Issue by xav-git** [fortinetdev/terraform-provider-fortios#299]
Hello, 

We have identified an issue with the resource "fortios_ips_sensor" running with provider ```v.1.17.0```.
When the resource is defined just like in the snapshot below, although it is correctly pushed to  the FortiGate on the first ```plan``` any successive ```plan``` or ```apply``` will report unnecessary changes. Please configuration and terraform output below describing the issues:

 fortios_ips_sensor definition in terraform: 
``` 
resource fortios_ips_sensor "corp-client-server-protect" { 
    vdomparam = "corp01"
    name = "corp-client-server-protect"
    comment = "block medium and above - alert below"


    entries {
         id = 1
         location = "client"
         severity = "medium high critical"
         #  Pass or drop matching traffic, depending on the default action of the signature.
         action = "default" 
    }
    entries {
         id = 2
         location = "server"
         severity = "medium high critical"
         #  Pass or drop matching traffic, depending on the default action of the signature.
         action = "default"  
    }
    entries {
         id = 3
         location = "client"
         severity = "info low"
         # Pass or allow matching traffic. same as Monitor
         action = "pass" 
    }
    entries {
         id = 4
         location = "server"
         severity = "info low"
         # Pass or allow matching traffic. same as Monitor
         action = "pass" 
    }
}
```

IPS Sensor usage
```
resource "fortios_firewall_policy" "dummy_ips_policy" {
  vdomparam = "corp01"
 [...]
 ips_sensor = "corp-client-server-protect"
```

Example of additional ```terraform plan / apply``` after a first successfull ```terraform apply```
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # fortios_ips_sensor.corp-client-server-protect will be updated in-place
  ~ resource "fortios_ips_sensor" "corp-client-server-protect" {
        id                      = "corp-client-server-protect"
        name                    = "corp-client-server-protect"
        # (7 unchanged attributes hidden)

      ~ entries {
            id                 = 1
          ~ location           = "client " -> "client"
            # (18 unchanged attributes hidden)
        }
      ~ entries {
            id                 = 2
          ~ location           = "server " -> "server"
            # (18 unchanged attributes hidden)
        }
      ~ entries {
            id                 = 3
          ~ location           = "client " -> "client"
            # (18 unchanged attributes hidden)
        }
      ~ entries {
            id                 = 4
          ~ location           = "server " -> "server"
            # (18 unchanged attributes hidden)
        }
    }

``` 

Can you please have a look at this issue ?
Thanks.

**Comment by vincent-at-forti** [comment]
Hello,

The problem lies on the fact that FortiOS REST API entry GET https://<IP>/api/v2/cmdb/ips/sensor/<name> returns entries containing extra space for location values.
```
{
  "http_method":"GET",
  "revision":"1127b4563a1f81d2ffa3e31cf7d5189c",
  "results":[
    {
      "name":"corp-client-server-protect",
      "q_origin_key":"corp-client-server-protect",
      "comment":"block medium and above - alert below",
      "replacemsg-group":"",
      "block-malicious-url":"disable",
      "scan-botnet-connections":"disable",
      "extended-log":"disable",
      "entries":[
        {
          "id":1,
          "q_origin_key":1,
          "rule":[
          ],
          "location":"client ",
          "severity":"medium high critical ",
          [...]
        },
        {
          "id":2,
          "q_origin_key":2,
          "rule":[
          ],
          "location":"server ",
          "severity":"medium high critical ",
          [...]
        },
        {
          "id":3,
          "q_origin_key":3,
          "rule":[
          ],
          "location":"client ",
          "severity":"info low ",
         [...]
        },
        {
          "id":4,
          "q_origin_key":4,
          "rule":[
          ],
          "location":"server ",
          "severity":"info low ",
          [...]
      ]
    }
  ],
  "vdom":"root",
  "path":"ips",
  "name":"sensor",
  "mkey":"corp-client-server-protect",
  "status":"success",
  "http_status":200,
  "serial":"FGVM02...",
  "version":"v7.2.5",
  "build":1517
}
```
We have reported this issue.

I noticed in the payload that severity also had same extra space.

But the fortios provider is making a workaround thanks to flattenIpsSensorEntriesSeverity in resource_ips_sensor.go file:
```
func flattenIpsSensorEntriesSeverity(v interface{}, d *schema.ResourceData, pre string, sv string) interface{} {
	if temp_v, ok := v.(string); ok {
		temp_v = strings.TrimRight(temp_v, " ")
		var rst_v interface{} = temp_v
		return rst_v
	}
	return v
}
```

So while waiting for FortiOS REST API fix, maybe we could add the same workaround for Locations:
```
func flattenIpsSensorEntriesLocation(v interface{}, d *schema.ResourceData, pre string, sv string) interface{} {
	if temp_v, ok := v.(string); ok {
		temp_v = strings.TrimRight(temp_v, " ")
		var rst_v interface{} = temp_v
		return rst_v
	}
	return v
}
```
instead of 
```
func flattenIpsSensorEntriesLocation(v interface{}, d *schema.ResourceData, pre string, sv string) interface{} {
	return v
}
```

Best Regards

**Comment by lix-fortinet** [comment]
Hi @xav-git,

This issue has been fixed since FortiOS Terraform v1.18.1. I will close this issue and feel free to open a new issue if you have any questions.

Thanks,
Xing
